frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewMetaData/EnvironmentVisibilityMenu/EnvironmentVisibilityMenu.tsx

```diff
@@ -0,0 +1,70 @@
+import { Button, Checkbox, Menu, MenuItem, styled } from '@mui/material';
+import { useState, type FC } from 'react';
+
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ExpandLessIcon from '@mui/icons-material/ExpandLess';
+import type { FeatureEnvironmentSchema } from 'openapi';
+
+type EnvironmentVisibilityMenuProps = {
+    environments: Pick<FeatureEnvironmentSchema, 'name'>[];
+    hiddenEnvironments: FeatureEnvironmentSchema['name'][];
+    onChange: (name: FeatureEnvironmentSchema['name']) => void;
+};
+
+const buttonId = 'environment-visibility-button';
+const menuId = 'environment-visibility-menu';
+
+const StyledContainer = styled('div')(({ theme }) => ({
+    display: 'flex',
+    justifyContent: 'center',
+    paddingTop: theme.spacing(4),
+}));
+
+export const EnvironmentVisibilityMenu: FC<EnvironmentVisibilityMenuProps> = ({
+    environments,
+    hiddenEnvironments,
+    onChange,
+}) => {
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+    const isOpen = Boolean(anchorEl);
+    const handleOpen = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    return (
+        <StyledContainer>
+            <Button
+                onClick={handleOpen}
+                endIcon={isOpen ? <ExpandLessIcon /> : <ExpandMoreIcon />}
+                variant='outlined'
+                id={buttonId}
+                aria-controls={isOpen ? menuId : undefined}
+                aria-haspopup='true'
+                aria-expanded={isOpen ? 'true' : undefined}
+                data-loading
+            >
+                Hide/show environments
+            </Button>
+            <Menu
```

Comment:
I'm not very familiar with the Menu component, so this may be a stupid question, but I just looked it up on the docs: do they autoclose on every selection? I would assume that you might want to show/hide multiple environments in one go with this, but maybe the auto-close on selection is an opt-in thing?

And out of my curiosity: is there a reason why you picked Menu over, say, Select?

Reply:
Select displays it's value in the button. If menu should close on every select is up to our implementation - in this case it will not.

Reply:
Sure; sounds sensible.